drivers/ieee802154/ieee802154_cc1101.c

```diff
@@ -0,0 +1,873 @@
+	}
+
+	return 0;
+}
+
+static inline u8_t get_rx_bytes(struct cc1101_context *ctx)
+{
+	u8_t rx_bytes;
+
+	if (!_cc1101_access_reg(ctx, true, CC1101_REG_RXBYTES,
+			       &rx_bytes, 1, true)) {
+		return 0;
+	}
+	
+	if(rx_bytes & 0x80) {
+		SYS_LOG_DBG("overflow\n");
+	}
+
+	return rx_bytes & 0x7F;
+}
+
+static inline bool verify_rxfifo_validity(struct cc1101_context *ctx,
+					  u8_t pkt_len)
+{
+	u8_t rx_bytes;
+
+	if (!_cc1101_access_reg(ctx, true, CC1101_REG_RXBYTES,
+			       &rx_bytes, 1, true)) {
+		return false;
+	}
+	
+	/* packet should be at least 3 bytes as a ACK */
+	if (pkt_len < 5 || rx_bytes > (pkt_len)) {
+		return false;
+	}
+
+	return true;
+}
+
+static inline bool read_rxfifo_content(struct cc1101_context *ctx,
+				       struct net_buf *frag, u8_t len)
+{
+
+	if (!read_rxfifo(ctx, frag->data, len) ||
+	    (get_status(ctx) == CC1101_STATUS_RXFIFO_OVERFLOW)) {
+		return false;
+	}
+
+	net_buf_add(frag, len);
+
+	return true;
+}
+
+static inline bool verify_crc(struct cc1101_context *ctx, 
+						struct net_pkt *pkt, u8_t len)
+{
+	u8_t *fcs = pkt->frags->data + len - CC1101_FCS_LEN;
+
+	if (!read_rxfifo(ctx, fcs, 2)) {
+		return false;
+	}
+
+	if (!(fcs[1] & CC1101_FCS_CRC_OK)) {
+		return false;
+	}
+
+	net_pkt_set_ieee802154_rssi(pkt, fcs[0]);
+	net_pkt_set_ieee802154_lqi(pkt, fcs[1] & CC1101_FCS_LQI_MASK);
+
+	return true;
+}
+
+static void cc1101_rx(struct device *dev)
+{
+	struct cc1101_context *cc1101 = dev->driver_data;
+	struct net_buf *pkt_frag;
+	struct net_pkt *pkt;
+	u8_t pkt_len;
+	u8_t status;
+	while (1) {
+		pkt = NULL;
+
+		k_sem_take(&cc1101->rx_lock, K_FOREVER);
+
+
+		status = get_status(cc1101);
+		_cc1101_print_status(status);
+		pkt_len = get_packet_length(cc1101);
+
+
+		SYS_LOG_DBG("red len: %d datalen %d\n", get_rx_bytes(cc1101), pkt_len);
+
+		pkt_len = get_rx_bytes(cc1101);
+
+		if (status == CC1101_STATUS_STARTCAL) {
+			SYS_LOG_ERR("start CAL error");
+			goto flush;
+		}
+
+		if (status == CC1101_STATUS_RXFIFO_OVERFLOW) {
+			SYS_LOG_ERR("RX FIFO OF error");
+			goto flush;
+		}
+
+		if (status == CC1101_STATUS_TXFIFO_UNDERFLOW) {
+			SYS_LOG_ERR("TX FIFO UF error");
+			goto flush;
+		}
+
+		pkt = net_pkt_get_reserve_rx(0, K_NO_WAIT);
+		if (!pkt) {
+			SYS_LOG_ERR("No free pkt available");
+			goto flush;
+		}
+
+		pkt_frag = net_pkt_get_frag(pkt, K_NO_WAIT);
+		if (!pkt_frag) {
+			SYS_LOG_ERR("No free frag available");
+			goto flush;
+		}
+
+		net_pkt_frag_insert(pkt, pkt_frag);
+
+		
+
+		if (!verify_rxfifo_validity(cc1101, pkt_len)) {
+			SYS_LOG_ERR("Invalid frame");
+			goto flush;
+		}
+
+		if (!read_rxfifo_content(cc1101, pkt_frag, pkt_len)) {
+			SYS_LOG_ERR("No content read");
+			goto flush;
+		}
+
+		if (!verify_crc(cc1101, pkt, pkt_len)) {
+			SYS_LOG_ERR("Bad packet CRC");
+			goto flush;
+		}
+
+		if (ieee802154_radio_handle_ack(cc1101->iface, pkt) == NET_OK) {
+			SYS_LOG_DBG("ACK packet handled");
+			goto flush;
+		}
+
+		SYS_LOG_DBG("rx: ");
+		int i;
```

Comment:
nope nope, never declare a variable in the middle of a code block: always a the beginning of the code block.